invoices/invoiceregistry.go

```diff
@@ -48,27 +48,32 @@ type HtlcResolution struct {
 
 	// AcceptHeight is the original height at which the htlc was accepted.
 	AcceptHeight int32
+
+	// Outcome indicates the outcome of the invoice registry update.
+	Outcome ResolutionResult
 }
 
 // NewHtlcResolution returns a non-settle htlc resolution.
 func NewHtlcResolution(key channeldb.CircuitKey,
-	acceptHeight int32) *HtlcResolution {
+	acceptHeight int32, outcome ResolutionResult) *HtlcResolution {
 
 	return &HtlcResolution{
 		CircuitKey:   key,
 		AcceptHeight: acceptHeight,
+		Outcome:      outcome,
```

Comment:
If you really like constructors, you could add a check here to see if `outcome` is a settle-specific outcome. Same for other constructor.

To not incur runtime overhead, it may also work by splitting up the `ResolutionResult` type in two (internal) types `SettleResult` and `FailResult`. The compiler can then catch misuse and in the constructor it can be cast into the more generic `ResolutionResult`.

But this is totally optional.

Reply:
I'm going to leave this one, I think the setting of results is sufficiently tested (ie, we're setting settles where we should) and the name/presence of preimage is a strong enough indicator for future code. 

I'd worry about setting up aliased types then casting in the constructor because we can't use iota across types so would have to declare mutually exclusive settle/fail result int values. Which would probably be fine, but would be a very nasty bug if it snuck in.

Reply:
Good point